File: 1914.py
P = int(input())
lista = []
for i in range(0, P):
    a, b, c, d = input().split()
    e, f = input().split()
    e = int(e)
    f = int(f)
    soma = e + f
    if soma % 2 == 0 and b == 'PAR':
        lista += [a]
    if soma % 2 == 0 and b == 'IMPAR':
        lista += [c]
    if soma % 2 != 0 and b == 'PAR':
        lista += [c]
    if soma % 2 != 0 and b == 'IMPAR':
        lista += [a]
for k in range(0, P):
    print(f'{lista[k]}')

File: 1914.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');

let P = parseInt(lines.shift());
let lista = [];
for(var i = 0; i < P; i++){
    let [a, b, c, d] = lines.shift().split(" ");
    let [e, f] = lines.shift().split(" ");
    e = parseInt(e); f = parseInt(f);
    let soma = e + f;
    if(soma % 2 === 0 && b === 'PAR'){
        lista.push(a);
    }
    if(soma % 2 === 0 && b === 'IMPAR'){
        lista.push(c);
    }
    if(soma % 2 !== 0 && b === 'PAR'){
        lista.push(c);
    }
    if(soma % 2 !== 0 && b === 'IMPAR'){
        lista.push(a);
    }
}
for(var k = 0; k < P; k++){
    console.log(`${lista[k]}`);
}
